ui/lib/kv/addon/components/page/secret/version-history.hbs

```diff
@@ -1,11 +1,102 @@
   </:tabLinks>
-
 </KvPageHeader>
+{{#if @metadata.canReadMetadata}}
+  {{#each @metadata.sortedVersions as |versionData|}}
+    <LinkedBlock
+      class="list-item-row"
+      @params={{array "vault.cluster.secrets.backend.kv.secret.details" @metadata.path}}
+      @queryParams={{hash version=versionData.version}}
+      data-test-version-linked-block={{versionData.version}}
+    >
+      <div class="level is-mobile">
+        <div class="is-grid is-grid-3-columns is-three-fourths-width">
+          {{! version number and icon }}
+          <div class="align-self-center">
+            <Icon @name="history" class="has-text-grey" data-test-version />
+            <span class="has-text-weight-semibold has-text-black">
+              Version
+              {{versionData.version}}
+            </span>
+          </div>
+          {{! icons }}
+          <div class="align-self-center" data-test-icon-holder={{versionData.version}}>
+            {{#if (loose-equal versionData.version @metadata.currentVersion)}}
+              <div>
+                <span class="has-text-success is-size-8 is-block">
+                  <Icon @name="check-circle-fill" />Current
+                </span>
+              </div>
+            {{/if}}
+            {{#if versionData.deletion_time}}
+              <div>
+                <span class="has-text-grey is-size-8 is-block">
+                  <Icon @name="x-square-fill" />Deleted
+                </span>
+              </div>
+            {{/if}}
+            {{#if versionData.destroyed}}
+              <div>
+                <span class="has-text-danger is-size-8 is-block">
+                  <Icon @name="x-square-fill" />Destroyed
+                </span>
+              </div>
+            {{/if}}
+          </div>
+          {{! version created date }}
+          <div class="is-size-8 has-text-weight-semibold has-text-grey align-self-center">
+            <ToolTip @verticalPosition="above" @horizontalPosition="center" as |T|>
+              <T.Trigger data-test-tooltip-trigger tabindex="-1">
+                Created
+                {{date-format versionData.created_time "MMM dd, yyyy hh:mm a"}}
+              </T.Trigger>
+              <T.Content @defaultClass="tool-tip smaller-font">
+                <div class="box" data-test-hover-copy-tooltip-text>
+                  {{versionData.created_time}}
+                </div>
+              </T.Content>
+            </ToolTip>
+          </div>
+        </div>
 
-<hr class="is-marginless has-background-gray-200" />
\ No newline at end of file
+        <div class="level-right">
+          <div class="level-item">
+            <PopupMenu @name="version-{{versionData.version}}">
+              <nav class="menu">
+                <ul class="menu-list">
+                  <li>
+                    <LinkTo @route="secret.details" @model={{@metadata.path}} @query={{hash version=versionData.version}}>
```

Comment:
Do we need to pass the model here? Mostly for my information - I can never remember with `LinkTo`. I know it's the same value - but I'd expect to see `@path` as the arg passed here instead of `@metadata.path`

Reply:
Good catch. I figured because it was a dynamic route that required the secret path I needed to pass in the model—ember doc's[ here ](https://guides.emberjs.com/release/routing/linking-between-routes/#toc_the-linkto--component)on when you pass in @model to LinkTo. But I'm guessing because it's a child route, I don't need to—it fills in that dynamic route for me.